front_end/audits2/Audits2Panel.js

```diff
@@ -1,43 +1,242 @@
+  }
+
+  /**
+   * @param {string} statusMessage
+   */
+  _updateStatus(statusMessage) {
+    this._statusElement.textContent = Common.UIString(statusMessage);
+  }
+
+  _stop() {
+    this._protocolService.detach().then(_ => {
+      this._auditRunning = false;
+      this._updateButton();
+      if (this._inspectedURL !== SDK.targetManager.mainTarget().inspectedURL())
+          SDK.targetManager.mainTarget().pageAgent().navigate(this._inspectedURL);
+
     });
   }
 
   /**
+   * @param {!Audits2.WorkerResult} result
+   */
+  _finish(result) {
+    this._stop();
+    this.resultsView.removeChildren();
+
+    this.resultsView.appendChild(this._createResultsBar(result.lighthouseResult.url, result.lighthouseResult.generatedTime));
+    this.resultsView.appendChild(this._createIframe(result.bloburl));
+    this.contentElement.classList.add('show-results');
+  }
+
+  /**
+   * @param {string} bloburl
+   * @return {!Element}
+   */
+  _createIframe(bloburl) {
+    var iframeContainer = createElementWithClass('div', 'iframe-container');
+    var iframe = iframeContainer.createChild('iframe', 'fill');
+    iframe.setAttribute('sandbox', 'allow-scripts allow-popups-to-escape-sandbox allow-popups');
+    iframe.src = bloburl;
+    return iframeContainer;
+  }
+
+  /**
+   * @param {string} url
+   * @param {string} timestamp
+   * @return {!Element}
+   */
+  _createResultsBar(url, timestamp) {
+    var elem = createElementWithClass('div', 'results-bar hbox');
+    elem.createChild('div', 'logo-small');
+
+    var summaryElem = elem.createChild('div', 'summary');
+    var reportFor = summaryElem.createChild('span');
+    reportFor.innerHTML = `Report for <b>${url}</b>`;
+    var timeElem = summaryElem.createChild('span');
+    timeElem.textContent =
+        `Generated at ${new Date(timestamp).toLocaleDateString()} ${new Date(timestamp).toLocaleTimeString()}`;
+
+    var newAuditButton = createTextButton(Common.UIString('New Audit'), this._reset.bind(this), 'new-audit audit-btn');
+    elem.appendChild(newAuditButton);
+    return elem;
+  }
+};
+
+Audits2.ProtocolService = class extends Common.Object {
+  constructor() {
+    super();
+    this._rawConnection = undefined;
+    this._backend = undefined;
+    this._backendPromise = undefined;
+    this._status = undefined;
+  }
+
+  /**
+   * @return {!Promise<undefined>}
+   */
+  attach() {
+    return SDK.targetManager.interceptMainConnection(this._dispatchProtocolMessage.bind(this))
+        .then(rawConnection => {
+          this._rawConnection = rawConnection;
+        });
+  }
+
+  /**
+   * @param {string} inspectedURL
+   * @return {!Promise<!Object|undefined>}
+   */
+  startLighthouse(inspectedURL) {
+      return this._send('start', {url: inspectedURL});
+  }
+
+  /**
+   * @return {!Promise<!Object|undefined>}
+   */
+  detach() {
+    return new Promise((resolve, reject) =>
+       this._send('stop').then(() => {
+        this._backend.dispose();
```

Comment:
> `WARNING - No properties on this expression`

looks like closure is only seeing where you define `this._backend` to be `undefined` in the constructor. You can add `/** @type {!ServiceManager.Service|undefined} */` over `this._backend = undefined;` in the constructor, but it might still think it could be `undefined` here due to the tricky work with `this._send` ensuring that it *is* defined, so you may have to still cast here.

(you can also wrap in a `if (this._backend) {}`, but since it will always be defined by this point, it's code just to make the compiler happy...better to just cast)